### PR TITLE
Include guide on HowToDoSerial

### PR DESCRIPTION
**Pull request by henrique-silva**
Original page:
https://epics-modules.github.io/master/asyn/R4-42/HowToDoSerial/HowToDoSerial_StreamDevice.html

### CODE REVIEW

File: guides/HowToDoSerial.md [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+I based this tutorial on the device support I wrote for a Hewlett Packard E3631A power supply. I chose the E3610A as the basis for this tutorial partly because it was what I found lying around but mostly because it uses SCPI-compliant commands which are common to a wide range of devices. Sections 1 through 4 of this tutorial apply to all SCPI (IEEE-488.2) devices.
+
+2 Create a new device support module
+------------------------------------
+
+The easiest way to do this is with the `makeSupport.pl` script supplied with the EPICS ASYN package.
+
+Here are the commands I ran. To make the command examples a little shorter I used shell variables to hold the paths to my installation of EPICS base and the ASYN support module. You can do this with values appropriate for your site or just type the full path names to the commands.
+
+```
```
Comment (gabrielfedel) [inline guides/HowToDoSerial.md:30]: I think you can put makefile syntax here

File: guides/HowToDoSerial.md [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+norume> mkdir HPE3631A
+norume> cd HPE3631A
+norume> $ASYN/bin/$EPICS_HOST_ARCH/makeSupport.pl -t streamSCPI HPE3631A
+```
+
+### 2.1 Make changes to some files in `configure/`
+
+Edit the `configure/RELEASE` file which `makeSupport.pl` created and confirm that the entries describing the paths to your EPICS base and ASYN support are correct. For example these might be:
+
+```
```
Comment (gabrielfedel) [inline guides/HowToDoSerial.md:49]: Here also

File: guides/HowToDoSerial.md [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+
+To install the support module files and build the test application simply run `make`:
+
+```console
+norume> make
+```
+
+The IOC startup script produced by the `makeBaseApp.pl` script must be modified before the test application can be run. Change directories to `iocBoot/iocHPE3631Atest` and edit the `st.cmd` file. Here's the file that I use:
+
+```
```
Comment (gabrielfedel) [inline guides/HowToDoSerial.md:137]: console syntax ?

File: guides/HowToDoSerial.md [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+dbLoadRecords("db/asynRecord.db","P=$(P)$(R),R=asyn,PORT=L0,ADDR=-1,OMAX=0,IMAX=0")
+```
+
+to the application startup script. The `PORT` value must match the the value in the corresponding `drvAsynIPPortConfigure` or `drvAsynSerialPortConfigure` command. The `ADDR` value should be that of the instrument whose I/O you wish to monitor.
+
+To run the asynRecord screen, add `<asynTop>/opi/medm` to your `EPICS_DISPLAY_PATH` environment variable and start medm with `P`, `R`, `PORT` and `ADDR` values matching those given in the dbLoadRecords command:
+
+```
```
Comment (gabrielfedel) [inline guides/HowToDoSerial.md:342]: bash syntax ?

File: guides/HowToDoSerial.md [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+medm -x -macro "P=hpE3631ATest,R=asyn,PORT=L0,ADDR=-1" asynRecord.adl
+```
+
+If you have edm installed then you need to use the `asynRecord.edl` file instead. Ensure that `<asynTop>/opi/medm` is in the `EDMDATAFILES` environment variable. Start edm with the arguments appropriate for your IOC:
+
+```
```
Comment (gabrielfedel) [inline guides/HowToDoSerial.md:348]: same here

File: guides/HowToDoSerial.md (hunk 1 above)
Comment (gabrielfedel) [reply guides/HowToDoSerial.md:30]: I found this: https://markdown.land/markdown-code-block
it show as make, but I think makefile should work also as I noticed on other files.

**Comment by gabrielfedel** [comment]
LGTM!

File: guides/HowToDoSerial.md [hunk 6]
```diff
@@ -0,0 +1,575 @@
+HowToDoSerial(StreamDevice)
+
```
[the remark this answers is not in the document]
Comment (henrique-silva) [reply guides/HowToDoSerial.md:2]: Oh, great catch, I didn't see that one.

I can put `HowToDoSerial(StreamDevice)` as H1 and `How to use StreamDevice and ASYN to create EPICS device support for a simple serial, GPIB, or network attached device` as H2, or just remove this first header altogether. Any preferences?

[the remark this answers is not in the document]
Comment (minijackson) [reply guides/HowToDoSerial.md:2]: I think I prefer removing `HowToDoSerial(StreamDevice)` altogether.

The first H1 title is the title that's going to be displayed on the sidebar, so maybe we should pick something a title less long than " How to use StreamDevice and ASYN to create EPICS device support for a simple serial, GPIB, or network attached device"?

File: guides/HowToUseStreamDevice.md [hunk 7]
```diff
@@ -0,0 +1,648 @@
...
+ 22 # drvAsynSerialPortConfigure port ipInfo priority noAutoconnect noProcessEos
+ 23 drvAsynSerialPortConfigure("L0","$(TTY)",0,0,0)
+ 24 asynSetOption("L0", -1, "baud", "9600")
+ 25 asynSetOption("L0", -1, "bits", "8")
+ 26 asynSetOption("L0", -1, "parity", "none")
+ 27 asynSetOption("L0", -1, "stop", "2")
+ 28 asynSetOption("L0", -1, "clocal", "Y")
+ 29 asynSetOption("L0", -1, "crtscts", "Y")
+ 30 asynOctetSetInputEos("L0", -1, "\\n")
+ 31 asynOctetSetOutputEos("L0", -1, "\\n")
```
Comment (simon-ess) [inline guides/HowToUseStreamDevice.md:224]: I have to check: are these escapes correct? We have had a lot of extra escape characters sneaking in...

File: guides/HowToUseStreamDevice.md [hunk 8]
```diff
@@ -0,0 +1,648 @@
...
+# drvAsynSerialPortConfigure port ipInfo priority noAutoconnect noProcessEos
+drvAsynSerialPortConfigure("L0","/dev/tty.PL2303-000013FA",0,0,0)
+asynSetOption("L0", -1, "baud", "9600")
+asynSetOption("L0", -1, "bits", "8")
+asynSetOption("L0", -1, "parity", "none")
+asynSetOption("L0", -1, "stop", "2")
+asynSetOption("L0", -1, "clocal", "Y")
+asynSetOption("L0", -1, "crtscts", "Y")
+asynOctetSetInputEos("L0", -1, "\\n")
+asynOctetSetOutputEos("L0", -1, "\\n")
```
Comment (simon-ess) [inline guides/HowToUseStreamDevice.md:315]: Same here.

File: guides/HowToUseStreamDevice.md [hunk 9]
```diff
@@ -0,0 +1,648 @@
...
+### 5.3 - Add records to set and check the output voltage and current setpoints
+
+Add two entries to `HPE3631ASup/devHPE3631A.proto`:
+
+```
+setVI {
+    out "INST \\$1;\\$2 %g";
+}
+getVI {
+    out "INST \\$1;\\$2?";
```
Comment (simon-ess) [inline guides/HowToUseStreamDevice.md:489]: I would confirm the escape characters here too.

File: guides/HowToUseStreamDevice.md (hunk 7 above)
Comment (ttkorhonen) [reply guides/HowToUseStreamDevice.md:224]: The colleagues here at codeathon agree that a single backslash should be sufficient.

**Comment by ttkorhonen** [comment]
I took the changes to HowToDoStreamDevice and put that in my local fork. Because the directory structure has changed, I close this PR and create a new one; first with only the file, and add it to the navigation later as I am working to re-organize the navigation menus.

**Pull request closed without merging**